featuretools/tests/computational_backend/test_calculate_feature_matrix.py

```diff
@@ -2266,3 +2275,25 @@ def multi_cum_sum(x):
     for col in feature_matrix.columns:
         origin = feature_matrix.ww[col].ww.origin
         assert origin in ["base", "engineered"]
+
+
+def test_renamed_features_have_expected_column_names(pd_es):
+    direct_feat = ft.Feature(pd_es["sessions"].ww["device_name"], "log")
+    trans_feat = ft.Feature(pd_es["log"].ww["value"], primitive=Negate)
+    multi_output_agg_feat = ft.Feature(
+        pd_es["log"].ww["product_id"],
+        parent_dataframe_name="customers",
+        primitive=NMostCommon(n=2),
+    )
+
+    direct_name = ["device_direct"]
+    direct_feat.set_feature_names(direct_name)
```

Comment:
for single output features, do we want `set_feature_names` and `rename` to both work?

Reply:
Not sure - are you suggesting that we just allow `set_feature_names` to work for multi-output features and renaming feature columns for single output should be done through `rename` as you can be done currently?

Reply:
As I think about it I think I like the idea of having `set_feature_names` operate only on multi-output features - seems more consistent with the method name, and would reduce the scope of the change. We could add a check in the method to raise an error if `number_output_features == 1`.

@rwedge What do you think about that approach?

Reply:
@rwedge Sounds good. I'll make this change and add the additional test case you recommended.